src/participant/participant.ts

```diff
@@ -333,40 +262,52 @@ export default class ParticipantController {
     }
   }
 
-  async _selectDatabaseWithCommandPalette(): Promise<string | undefined> {
+  async _selectDatabaseWithQuickPick(): Promise<string | undefined> {
     const databases = await this.getDatabaseQuickPicks();
     const selectedQuickPickItem = await vscode.window.showQuickPick(databases, {
       placeHolder: 'Select a database...',
     });
     return selectedQuickPickItem?.data;
   }
 
-  async selectDatabaseWithParticipant(name: string): Promise<boolean> {
-    if (!name) {
-      const selectedName = await this._selectDatabaseWithCommandPalette();
-      this._setDatabaseName(selectedName);
-    } else {
-      this._setDatabaseName(name);
+  async selectDatabaseWithParticipant({
+    chatId,
+    databaseName: _databaseName,
+  }: {
+    chatId: string;
+    databaseName: string;
+  }): Promise<boolean> {
+    let databaseName: string | undefined = _databaseName;
+    if (!databaseName) {
+      databaseName = await this._selectDatabaseWithQuickPick();
+      if (!databaseName) {
+        return false;
+      }
     }
 
+    this._chatMetadataStore.setChatMetadata(chatId, {
+      databaseName: databaseName,
+    });
     return vscode.commands.executeCommand('workbench.action.chat.open', {
-      query: `@MongoDB /query ${this._databaseName || ''}`,
+      // Add a database name message to the chat so future messages know which namespace to use.
+      query: `@MongoDB /query ${databaseName}`,
```

Comment:
There is no other way to tell the model to proceed with the query generation flow. When a response is printed to the chat, only another prompt triggers the request handler. I would actually remove the comment since it is a bit misleading. Especially now when we store a namespase in the dedicated storage.

Reply:
Added a helper function for this, with a comment as to why we do it. As Alena said, it's something we have to do for activating the request handler.